server/storage/backend/read_tx.go

```diff
@@ -32,14 +33,19 @@ type ReadTx interface {
 	RUnlock()
 
 	UnsafeRange(bucket Bucket, key, endKey []byte, limit int64) (keys [][]byte, vals [][]byte)
+	UnsafeRangeWithLock(bucketType Bucket, key, endKey []byte, limit int64, b Backend) (keys [][]byte, vals [][]byte)
```

Comment:
Can we use a more precise name for this function? such as UnsafeRangeSharedBuffer()